narwhals/_polars/series.py

```diff
@@ -120,10 +121,14 @@ def dtype(self: Self) -> DType:
     def __getitem__(self: Self, item: int) -> Any: ...
 
     @overload
-    def __getitem__(self: Self, item: slice | Sequence[int]) -> Self: ...
-
-    def __getitem__(self: Self, item: int | slice | Sequence[int]) -> Any | Self:
-        return self._from_native_object(self._native_series.__getitem__(item))
+    def __getitem__(self: Self, item: slice | Sequence[int] | NWSeries[Any]) -> Self: ...
+
+    def __getitem__(
+        self: Self, item: int | slice | Sequence[int] | NWSeries[Any]
+    ) -> Any | Self:
+        if isinstance(item, (int, slice, Sequence)):
+            return self._from_native_object(self._native_series.__getitem__(item))
+        return self._from_native_object(self._native_series.__getitem__(item.to_numpy()))
```

Comment:
🤔 not sure we should be converting to numpy here

Reply:
Thank you, I had doubts about this too and picked a solution that seemed to work without actually understanding the consequences.

Reply:
It seems using `item.to_native()` instead works, and also solves an issue with missing values becoming `NaN` with `to_numpy()`. Do you think it's acceptable? Any other corner cases to cover with tests?

Reply:
thanks - i'll check, to be careful we get the details right here